### PR TITLE
Detect multiprocess environment and use MultiProcessCollector

### PR DESCRIPTION
**Pull request by elephantum**

### CODE REVIEW

**Comment by rycus86** [comment]
Hey! Thanks a lot for the PR!

Do you happen to know whether this might solve the issues reported in https://github.com/rycus86/prometheus_flask_exporter/issues/5 ?

Also, just nit picking, but could we perhaps keep the function as a single one and only `if/else` the bits that are different?

Thanks!

**Comment by elephantum** [comment]
@rycus86 yes, this PR is targeted to fix #5, at least partially.

This PR makes standard metrics to be aggregate of all processes. (As opposed to `/metrics` serving only part of information).

More thought should be put into dealing with Gauges as they require additional configuration to work correctly in multiprocessing env. I did not address it.

**Comment by rycus86** [comment]
Great, thanks! Let me have a play with it tomorrow, then I think it can go in and I'll do a release to PyPI.

File: prometheus_flask_exporter/__init__.py [hunk 1]
```diff
@@ -109,10 +111,17 @@ def register_endpoint(self, path, app=None):
         @app.route(path)
         @self.do_not_track()
         def prometheus_metrics():
-            registry = self.registry
+            if 'prometheus_multiproc_dir' in os.environ:
+                registry = CollectorRegistry()
```
[the remark this answers is not in the document]
Comment (elephantum) [reply prometheus_flask_exporter/__init__.py]: Yes.

The thing with multiprocess metrics is that every process writes several *.db files to `prometheus_multiproc_dir`, then during `/metrics` handling all of them are read, aggregated and rendered.

So we don't really need this specific process' registry it is used to create one of the files, not to render `/metrics`.

[the remark this answers is not in the document]
Comment (rycus86) [reply prometheus_flask_exporter/__init__.py]: Awesome, thanks for the explanation!

> More thought should be put into dealing with Gauges as they require additional configuration to work correctly in multiprocessing env. I did not address it.

Can you give a bit more info on this? What's needed/missing here?

I'll merge this in now, and release it, then follow up on any further changes if needed.

Thanks!

[the remark this answers is not in the document]
Comment (rycus86) [reply prometheus_flask_exporter/__init__.py]: Oh, just found https://github.com/prometheus/client_python#multiprocess-mode-gunicorn again.
Were you referring to this with regards to Gauges?

> Gauges have several modes they can run in, which can be selected with the multiprocess_mode parameter.
>
> - 'all': Default. Return a timeseries per process alive or dead.
> - 'liveall': Return a timeseries per process that is still alive.
> - 'livesum': Return a single timeseries that is the sum of the values of alive processes.
> - 'max': Return a single timeseries that is the maximum of the values of all processes, alive or dead.
> - 'min': Return a single timeseries that is the minimum of the values of all processes, alive or dead.

[the remark this answers is not in the document]
Comment (elephantum) [reply prometheus_flask_exporter/__init__.py]: Yes, that's it.

**Pull request merged**